scripts/generate_layers_metrics.py

```diff
@@ -0,0 +1,358 @@
+    """
+    layer_stack = []
+    pt_compile_model_time = time.time()
+
+    module_depth = {}
+    module_name = {}
+
+    def register_depths(module, current_depth=0, name='model'):
+        module_depth[module] = current_depth
+        module_name[module] = name
+        parent=name
+        # if we are dealing with array of layers
+        array_layers = all(key.isdigit() for key in module._modules.keys())
+        for name, child in module._modules.items():
+            if array_layers: 
+                register_depths(child, current_depth + 1, parent+'['+name+']')
+            else:
+                register_depths(child, current_depth + 1, parent+'.'+name)
+
+    register_depths(model)
+
+    def wrap_forward(layer):
+        original_forward = layer.forward
+
+        def safe_forward(*args, **kwargs):
+            try:
+                return original_forward(*args, **kwargs)
+            except (RuntimeError,TypeError) as e:
+                logger.error(f"Error in {layer.__class__.__name__}: {e}")
+                return torch.zeros_like(args[0]) if args else None
+        layer.forward = safe_forward
+        
+
+    hooks = []
+    def pre_hook_fn(module, input):
+        depth = module_depth.get(module, 0)
+        layer_name = module_name.get(module, 0)
+        prefix = '│    ' * depth
+        if len(input) == 0: return
+        input_shape_str = f"[{', '.join(map(str, input[0].shape))}]"
+        input_type = str(input[0].dtype)
+        if module.parameters() == None: return
+        param_size = sum(p.numel() for p in module.parameters() if p.requires_grad)
+        param_size_str = f"{param_size:,}" if param_size > 0 else "--"
+        logger.info(f"{prefix}├─{layer_name}() -> {module.__class__.__name__} : | Input(arg): {input_shape_str} | {input_type} | Params: {param_size_str}")
+        wrap_forward(module)
+        # save input for later use with outputs
+        module._debug_input = input 
+
+    def post_hook_fn(module, input, output):
+        layer_name = module_name.get(module, 0)
+        # Save inputs and outputs
+        if hasattr(module, '_debug_input'):
+            layer_stack.append((layer_name, output))
+            # Clean up
+            delattr(module, '_debug_input')
+    
+    for name, layer in model.named_modules():
+        hooks.append(layer.register_forward_pre_hook(pre_hook_fn))
+        hooks.append(layer.register_forward_hook(post_hook_fn))
+
+    
+    __infer_layer(model= model, max_len=seq_length, 
+                  device=device, max_new_tokens=max_new_tokens, 
+                  batch_size=batch_size, tokenizer=tokenizer)
+
+    for hook in hooks:
+        hook.remove()
+
+    pt_compile_model_time = time.time() - pt_compile_model_time
+    logger.info(f"PT compile complete, took {pt_compile_model_time:.3f}s")
+
+    return layer_stack
+
+def write_csv(l, path, metric, gpu_layer_shape, cpu_layer_shape, output_shape):
```

Comment:
The variable name `l` is ambiguous, non-descriptive, and visually similar to the numeral 1. can you replace it with something like layer_output_list or similar for clarity.